### PR TITLE
fix(chartutil): move values coalescing into chartutil

### PR DESCRIPTION
**Pull request by technosophos**
This separates values file processing from template rendering. This will make it easier for us to...
- Display/store just the rendered values
- Implement alternative template engines
- Pre-compute coalesced values for linting and client verification

### CODE REVIEW

**Comment by technosophos** [comment]
Still to finish:

- [] Re-work passing the rendered map into the Engine
- [] Finish implementing global variables

**Comment by technosophos** [comment]
See also #767

File: pkg/chartutil/values_test.go [hunk 1]
```diff
@@ -139,3 +142,106 @@ func ttpl(tpl string, v map[string]interface{}) (string, error) {
...
+	}
+
+	if addr["state"].(string) != "MA" {
+		t.Errorf("Unexpected state: %v", addr["state"])
+	}
+
+	if det, ok := dst["details"].(map[string]interface{}); !ok {
+		t.Fatalf("Details is the wrong type: %v", dst["details"])
+	} else if _, ok := det["friends"]; !ok {
+		t.Error("Could not find your friends. Maybe you don't have any. :-(")
```
Comment (adamreese) [inline pkg/chartutil/values_test.go]: I'll be your friend

Comment (technosophos) [inline pkg/chartutil/values_test.go]: I'm so glad somebody reads my tests.

**Pull request merged**